### PR TITLE
Make it possible to build from local sources

### PR DESCRIPTION
**Pull request by antoine-de**
several addition for dev purpose:

* add a dev image to build navitia and run tests
* add the possibility to build all the images from the local sources
* add the possibility to run jormungandr with the local sources

Note: I did not manage to make the uwsgi hot reload work :confused:

### CODE REVIEW

File: builder/build.sh [hunk 1]
```diff
@@ -29,7 +30,7 @@ user=''
 password=''
 components='jormungandr kraken tyr-beat tyr-worker tyr-web'
 
-while getopts "lrb:u:p:" opt; do
+while getopts "lrnb:u:p:" opt; do
```
Comment (kinnou02) [inline builder/build.sh]: it should be `n:` if it's the path to the navitia directory

File: builder/build.sh [hunk 2]
```diff
@@ -55,19 +59,28 @@ done
...
+    popd
+fi
 
-git describe
+run cmake -DCMAKE_BUILD_TYPE=Release $navitia_dir/source
+run make -j$(nproc) kraken ed_executables protobuf_files
 
+pushd $navitia_dir
 version=$(git describe)
+echo "building version $version"sqp
```
Comment (kinnou02) [inline builder/build.sh]: `sqp`?

File: builder/build.sh [hunk 3]
```diff
@@ -55,19 +59,28 @@ done
...
+popd
 
 for component in $components; do
-    run docker build -t navitia/$component:$version -f  Dockerfile-$component .
+    run docker build -t navitia/$component:$version -f  Dockerfile-$component . 
     if [ $tag_latest -eq 1 ]; then
```
Comment (kinnou02) [inline builder/build.sh]: trailing space

File: dev/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# Docker image used to build and run navitia's test
+FROM debian:8
```
Comment (kinnou02) [inline dev/Dockerfile]: you should derive from navitia/debian8_dev

**Comment by antoine-de** [comment]
done

**Comment by antoine-de** [comment]
done

**Pull request merged**